### PR TITLE
Ports/libtheora: Change download link to http instead of https

### PR DESCRIPTION
**Pull request by EWouters**
Fixes download on Mac where an old version of LibreSSL won't redirect to the download mirror.

### CODE REVIEW

**Comment by bcoles** [comment]
Can we download it from somewhere else that does support SSL? Perhaps the GitLab repo?

https://gitlab.xiph.org/xiph/theora

Presumably `http://downloads.xiph.org` redirects to the closest mirror.

**Comment by EWouters** [comment]
> Can we download it from somewhere else that does support SSL? Perhaps the GitLab repo?
> 
> https://gitlab.xiph.org/xiph/theora
> 
> Presumably `http://downloads.xiph.org` redirects to the closest mirror.

We discussed this on discord in #ports, the download will still happen over SSL from the mirror, just the redirect will be handled by the http site of the main mirror.

**Comment by EWouters** [comment]
The gitlab repo contains different source code.

**Comment by bcoles** [comment]
**tl/dr**: IMO the easiest and laziest option is to simply use the `ftp.osuosl.org` mirror directly.

> The gitlab repo contains different source code.

Welp.

>We discussed this on discord in #ports, the download will still happen over SSL from the mirror, just the redirect will be handled by the http site of the main mirror.

We could use the mirror directly, but that's not ideal either. I'm not sure how many mirrors there are, but I was redirected to the same mirror `https://ftp.osuosl.org/pub/xiph/releases/theora/` so it probably does not redirect based on location.

`ftp.osuosl.org` (140.211.166.134) is also located in the same netblock as `downloads.xiph.org` (140.211.166.31) so it is possible they have only one mirror.

> Fixes download on Mac where an old version of LibreSSL won't redirect to the download mirror.

Taking a look at the SSL configuration of `downloads.xiph.org` (140.211.166.31) shows the server supports only TLS1.3.

* https://www.ssllabs.com/ssltest/analyze.html?d=downloads.xiph.org&s=140.211.166.31

Where as the `ftp.osuosl.org` mirror (140.211.166.134) supports only TLS1.2.

* https://www.ssllabs.com/ssltest/analyze.html?d=ftp.osuosl.org&s=140.211.166.134&latest

It seems like LibreSSL has been discussing support for TLS1.3 for 5 years but still haven't implemented it. This is starting to cause problems for package maintainers and developers who are dropping support for LibreSSL.

* https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=255677
* https://www.gentoo.org/support/news-items/2021-01-05-libressl-support-discontinued.html

The arguing in this thread only raises more questions than it answers:

* https://apple.stackexchange.com/questions/324222/how-official-is-libressl-support-in-high-sierra

If the situation continues, Mac users are going to have more pressing issues than being unable to install Serenity ports.

FWIW; Serenity does not yet support SSL/TLS. A simple hack is used when installing ports inside Serenity to use HTTP instead:

https://github.com/SerenityOS/serenity/blob/98accad9baf74c9c1f56621ff5ae7dbbd1f54ec6/Ports/.port_include.sh#L214-L217

I'd prefer we use HTTPS rather than HTTP where possible. I wrote about this previously (https://github.com/SerenityOS/serenity/pull/6210#issuecomment-817257626) but it is a somewhat unpopular opinion and difficult to justify given that we validate the SHA256 hash.

The question is, is the tradeoff for supporting pre-TLS1.3 clients (Windows 7, MacOS apparently, and anything using LibreSSL) worth the minor and unlikely consequences of using clear text HTTP?

**Comment by EWouters** [comment]
Changed to `https://ftp.osuosl.org/pub/xiph/releases/theora/` mirror.

**Review by bcoles — APPROVED** [review]

**Comment by linusg** [comment]
> but it is a somewhat unpopular opinion

Is it? preferring HTTPS over HTTP seems like a pretty popular opinion in this day and age :^)

> and difficult to justify given that we validate the SHA256 hash.

Sure, there's some overlap, but IMO they serve slightly different purposes:

- The hash ensures that we received what we expected to receive
- HTTPS ensures that both the request and response (...path, headers, body) weren't altered *or* viewed in transit, so it adds additional value

**Comment by bcoles** [comment]
> Sure, there's some overlap, but IMO they serve slightly different purposes:

Yes. See: https://github.com/SerenityOS/serenity/pull/6210#issuecomment-817257626
